pkg/config/instrumentation/config_metrics.go

```diff
@@ -0,0 +1,59 @@
+package instrumentation
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type configMetrics struct {
+	configHash          *prometheus.GaugeVec
+	configReloadSuccess prometheus.Gauge
+	configReloadSeconds *prometheus.GaugeVec
+}
+
+// ConfigMetrics exposes metrics related to configuration loading
+var ConfigMetrics = newConfigMetrics()
+
+func newConfigMetrics() *configMetrics {
+	var m configMetrics
+
+	m.configHash = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "agent_config_hash",
+			Help: "Hash of the currently active config file.",
+		},
+		[]string{"sha256"},
+	)
+	m.configReloadSuccess = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "agent_config_last_reload_successful",
```

Comment:
question: shall we be using the term `load` instead of `reload` here? Reload could be confusing given that we instrument every load, including the initial one

Reply:
Sounds reasonable.